src/vaadin-grid-scroll-mixin.html

```diff
@@ -386,14 +386,20 @@
         this.$.footer.style.transform = this.$.header.style.transform = this._getTranslate(0, this._scrollTop);
       }
 
-      var frozenCellTransform = this._getTranslate(this._scrollLeft, 0);
+      var frozenCellTransform = this._getTranslate(this._scrollLeft, 0, true);
       for (var i = 0; i < this._frozenCells.length; i++) {
         this._frozenCells[i].style.transform = frozenCellTransform;
       }
     }
 
-    _getTranslate(x, y) {
-      return 'translate(' + x + 'px,' + y + 'px)';
+    _getTranslate(x, y, rtl) {
+      const outerScroller = this.$.outerscroller;
+      const isRTL = this.getAttribute('dir') === 'rtl';
+
+      if (isRTL && rtl) {
+        x = - (outerScroller.scrollWidth - (x + outerScroller.clientWidth));
```

Comment:
Considering how often this gets invoked, getting the "dir" attribute from the dom element and calculating `(outerScroller.scrollWidth - outerScroller. clientWidth)` every time might cause some performance impact. Also, instead of adding a boolean parameter to the function, would be cleaner to leave the function as it is and add to the value that is passed to the function as x on line 389.

Reply:
Yeah, in any case the calculation will most probably have performance impact. I am still searching for the proper place (i.e. overriding getters and calculations from `iron-scroll-target-behavior` with `rtl` ), it's just a "working fallback".